File: TwoPointers/reverse_only_letters.py
def reverse_only_letters(s: str) -> str:
    left, right = 0, len(s) - 1
    chars = list(s)

    while left < right:
        is_alpha_left = chars[left].isalpha()
        is_alpha_right = chars[right].isalpha()

        if is_alpha_left and is_alpha_right:
            chars[left], chars[right] = chars[right], chars[left]

            left += 1
            right -= 1
        elif not is_alpha_left:
            left += 1
        else:
            right -= 1

    return "".join(chars)
        
print(reverse_only_letters("a-bC-dEf-ghIj"))

File: Arrays&Strings/TwoPointers/reverseOnlyLetters.js
/**
 * @param {string} s 
 */
function reverseOnlyLetters(s) {
    let left = 0;
    let right = s.length - 1;
    const chars = s.split('');

    while (left < right) {
        const isLeftAlpha = isAlpha(chars[left]);
        const isRightAlpha =isAlpha(chars[right]);

        if (isLeftAlpha && isRightAlpha) {
            [chars[left], chars[right]] = [chars[right], chars[left]]
            left++;
            right--; 
        } else if (!isLeftAlpha) {
            left++;
        } else {
            right--;
        }
    }

    return chars.join('');

    /**
     * @param {string} char 
     */
    function isAlpha(char) {
        return /[a-zA-Z]/.test(char);
    }
}

console.log(reverseOnlyLetters("a-bC-dEf-ghIj"))